application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,319 @@
+
+    private static final Color AMBIENT_COLOR = new Color(24, 221, 136, 255);
+
+    private static final List<OptionData> messageOptions = List.of(
+            new OptionData(OptionType.STRING, TITLE_OPTION, "The title for the message", false),
+            new OptionData(OptionType.STRING, DESCRIPTION_OPTION, "A description for the message",
+                    false));
+
+
+    /**
+     * Construct an instance
+     */
+    public RoleSelectCommand() {
+        super("role-select", "Sends a message where users can select their roles",
+                SlashCommandVisibility.GUILD);
+
+        SubcommandData allRoles =
+                new SubcommandData(ALL_OPTION, "Lists all the rolls in the server for users")
+                    .addOptions(messageOptions);
+
+        SubcommandData selectRoles =
+                new SubcommandData(CHOOSE_OPTION, "Choose the roles for users to select")
+                    .addOptions(messageOptions);
+
+        getData().addSubcommands(allRoles, selectRoles);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+        if (!member.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+        boolean ephemeral = false;
+
+        if (Objects.equals(event.getSubcommandName(), CHOOSE_OPTION)) {
+            addMenuOptions(event, menu, "Select the roles to display", 1);
+            ephemeral = true;
+        } else {
+            addMenuOptions(event, menu, "Select your roles", 0);
+        }
+
+        // Handle Optional arguments
+        OptionMapping titleOption = event.getOption(TITLE_OPTION);
+        OptionMapping descriptionOption = event.getOption(DESCRIPTION_OPTION);
+
+        String title = handleOption(titleOption);
+        String description = handleOption(descriptionOption);
+
+        MessageBuilder messageBuilder = new MessageBuilder(makeEmbed(title, description))
+            .setActionRows(ActionRow.of(menu.build()));
+
+        if (ephemeral) {
+            event.reply(messageBuilder.build()).setEphemeral(true).queue();
+        } else {
+            event.getChannel().sendMessage(messageBuilder.build()).queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
```

Comment:
I might be missing something but why this split? Dont I see the message directly in the same channel or are there 2 different channels?

Reply:
The reply is for the slash-command, I don't want the reaction-role message to say that it's from a slash-command.

Thus I send the role-selection in a message beforehand.